api/src/main/java/org/openmrs/module/fhir2/model/FhirReference.java

```diff
@@ -50,6 +50,7 @@ public FhirReference() {
 	@Column(name = "target_uuid")
 	private String targetUuid;
 	
+	@EqualsAndHashCode.Include
```

Comment:
I'd prefer if we didn't do this. Honestly, I'm not even all that keen on having the `id` field annotated this way.

```suggestion
```

Reply:
@ibacher removing this annoation makes things fail .
it literally means the object hashcode is simply generated based on the Object [Id](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/model/FhirReference.java#L41).
and given that id is null before the object is persisted , the [set](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/TaskTranslatorImpl.java#L151-L152) would assume all these Reference Objects are equls so would deduplicate the References into one

Reply:
The other alternative is to change the [Set](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/model/FhirTask.java#L85) to List

Reply:
> Then we really should just get rid of the other EqualsAndHashCode.Include and fall back to using the UUID hash code...

Alright , let get rid of them